desktop/src/main/java/org/destinationsol/desktop/SolDesktop.java

```diff
@@ -77,10 +133,33 @@ public static void main(String[] argv) {
                 new Thread(() -> CrashReporter.report(ex, logPath)).start();
             });
         }
+    }
 
-        new LwjglApplication(new SolApplication(), applicationConfig);
+    /**
+     * Set up window resolution.
+     *
+     * When flag {@link DebugOptions#EMULATE_MOBILE} is set, make the app window the size of mobile screen. Otherwise,
+     * load the window resolution from game options.
+     *
+     * @param applicationConfig App config to configure
+     * @param reader {@link SolFileReader} to read stored settings with.
+     */
+    private static void setScreenDimensions(LwjglApplicationConfiguration applicationConfig, MyReader reader) {
+        if (DebugOptions.EMULATE_MOBILE) {
+            applicationConfig.width = 640;
+            applicationConfig.height = 480;
+            applicationConfig.fullscreen = false;
+        } else {
+            GameOptions d = new GameOptions(false, reader);
+            applicationConfig.width = d.x;
+            applicationConfig.height = d.y;
+            applicationConfig.fullscreen = d.fullscreen;
+        }
     }
 
+    /**
+     * Provides the default, and so far only, implementation of SolFileReader.
```

Comment:
No need to speculate about temporary states like number of implementations of something :-) It could become inaccurate pretty easily